packages/headless-react/package.json

```diff
@@ -15,15 +15,20 @@
     "ssr"
   ],
   "license": "Apache-2.0",
-  "main": "dist/index.js",
+  "type": "module",
+  "exports": "./dist/index.js",
```

Comment:
Oh yeah good catch please do introduce conditional exports right out of the gate.
This considerably reduce the surface level API.

Reply:
Currently with `"exports": "./dist/index.js",` the API surface is restricted to the exports in index. 
1. Reg adding `types` it doesn't seem like it is supported with `exports`
> "Conditional exports provide a way to map to different paths depending on certain conditions. They are supported for both CommonJS and ES module imports."

`types` is listed under user conditions as a Community guideline
> "custom user conditions can be added with the --conditions flag:"
> "Condition strings other than the "import", "require", "node", "node-addons" and "default" conditions [implemented in Node.js core](https://nodejs.org/api/packages.html#conditional-exports) are ignored by default."

From [TS docs](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) it is not clear if it will use `types` under `exports` - even if it does I would expect it to follow the current behaviour of using `<entry point>.d.ts` by default. 

2. Reg adding `ssr` subpath - it is not yet clear if/when/how-much non-ssr utils will be added to the pkg. The pkg desc is currently`"React utilities for SSR (Server Side Rendering) with headless"`. So redirecting all current exports from the default path to `ssr` subpath seems unnecessary and inconvenient (to users). Would it be better to add a subpath for non-ssr utils if/when they are created? 

3. Reg 
> I'm wondering here whether we should export only defineSearchEngine or all SSR utilities.
Since @coveo/headless is a dependency, not a peer dependency, I'm tempted to say we should re-export all of @coveo/headless/ssr. I think it would also be more straightforward for implementers.

I was wondering about this as well. Agree it might be good to reexport. Done.

Reply:
Concerning the subpath, my gut feeling is still that we should either have a sub-package or rename the package to imply ssr in the name.

My reasoning is that, if we do end up having some client-side utilities, mixing them with the SSR utilities may induce some confusion for both implementers and ourselves when we add new exports. Additionally, we would have parity with the structure of exports in `@coveo/headless`.

The main counter-point I see is that it's kinda ugly to have a whole NPM package with just a sub-path export, but I don't think that's as significant of a problem.

Reply:
> Concerning the subpath, my gut feeling is still that we should either have a sub-package or rename the package to imply ssr in the name.

Fair, and I think given our goal, the package name is the answer IMO: it should be clear what is it for. but well, naming things is hard.

> Currently with "exports": "./dist/index.js", the API surface is restricted to the exports in index.

👍 as long as only index.js is accessible through import, all good 👌 

> From [TS docs](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html) it is not clear if it will use types under exports - even if it does I would expect it to follow the current behaviour of using <entry point>.d.ts by default (which in this case should work without an explicit entry for index.d.ts).

Fun stuff about TS: sometimes their release notes are the best doc around:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-7.html#packagejson-exports-imports-and-self-referencing.

Reply:
Added `ssr` package, moved files to `src/ssr` and updated headless-ssr examples.